runtime/near-vm-logic/src/tests/gas_counter.rs

```diff
@@ -350,6 +354,332 @@ fn create_promise_dependency(logic: &mut VMLogic) -> Result<(), VMLogicError> {
     Ok(())
 }
 
+#[test]
+fn out_of_gas_new_data_receipt_byte() {
+    check_action_gas_exceeds_limit(ActionCosts::new_data_receipt_byte, 11, value_return);
```

Comment:
I would add comments for `check_action_gas_exceeds_limit` and `check_action_gas_exceeds_attached` explaining the relations between `exercise_action` and other calls - e.g. "it is guaranteed that `exercise_action` triggers charging `cost` `num_action_paid` times".

Also we could have a comment in the beginning of the file explaining the general scheme of the tests:
* defining specific "exercise_action" which should trigger charging some specific cost
* call `check_action_gas_exceeds_limit`
* call `check_action_gas_exceeds_attached`

Reply:
good point, I added a bunch of new comments in the latest commit, please take a look